python_modules/dagster/dagster_tests/components_tests/test_definitions.py

```diff
@@ -90,3 +93,20 @@ def my_defs_with_context(context: ComponentLoadContext):
     assert len(assets) == 1
 
     assert check.inst(next(iter(assets)), AssetSpec).key.path[0] == "asset1"
+
+
+def test_component_tree():
+    dagster_test_path = Path(__file__).joinpath("../../../../dagster-test").resolve()
+    assert dagster_test_path.exists()
+    defs = ComponentTree.load(dagster_test_path).load_defs()
+
+    repo_snap = RepositorySnap.from_def(defs.get_repository_def())
+    assert repo_snap.component_tree
+
+    inst_map = {snap.key: snap.full_type_name for snap in repo_snap.component_tree.leaf_instances}
+    assert inst_map == {
+        "composites/python[first]": "dagster_test.dg_defs.composites.python.component.PyComponent",
+        "composites/python[second]": "dagster_test.dg_defs.composites.python.component.PyComponent",
+        "composites/yaml[0]": "dagster_test.components.simple_asset.SimpleAssetComponent",
```

Comment:
the automatic ordinal scheme here is a little unfortunate, but I guess worst case is urls in the UI etc might break?